tests/unit/test_apigateway.py

```diff
@@ -107,6 +114,41 @@ def test_apply_request_parameters(self):
         )
         self.assertEqual("https://httpbin.org/anything/foo/bar/baz?param=foobar", uri)
 
+    def test_if_request_is_valid_with_no_resource_methods(self):
+        ctx = ApiInvocationContext("POST", "/", b"", {})
+        validator = RequestValidator(ctx, None)
+        self.assertTrue(validator.is_request_valid())
+
+    def test_if_request_is_valid_with_no_matching_method(self):
+        ctx = ApiInvocationContext("POST", "/", b"", {})
+        ctx.resource = {"resourceMethods": {"GET": {}}}
+        validator = RequestValidator(ctx, None)
+        self.assertTrue(validator.is_request_valid())
+
+    def test_if_request_is_valid_with_no_validator(self):
+        ctx = ApiInvocationContext("POST", "/", b"", {})
+        ctx.api_id = "deadbeef"
+        ctx.resource = {"resourceMethods": {"POST": {"requestValidatorId": " "}}}
+        validator = RequestValidator(ctx, None)
+        self.assertTrue(validator.is_request_valid())
+
+    def test_if_request_has_body_validator(self):
+        apigateway_client = Mock(boto3.client(service_name="apigateway"))
+        apigateway_client.get_request_validator.return_value = {"validateRequestBody": True}
+        apigateway_client.get_model.return_value = {"schema": '{"type": "object"}'}
+        ctx = ApiInvocationContext("POST", "/", '{"id":"1"}', {})
+        ctx.api_id = "deadbeef"
+        ctx.resource = {
+            "resourceMethods": {
+                "POST": {
+                    "requestValidatorId": "112233",
+                    "requestModels": {"application/json": '{ "id": "1"}'},
+                }
+            }
+        }
+        validator = RequestValidator(ctx, apigateway_client)
+        self.assertTrue(validator.is_request_valid())
```

Comment:
nit: Can we also add a negative test here, where `validator.is_request_valid()` returns `False`?